arch/arm/src/armv7-r/arm_syscall.c

```diff
@@ -526,10 +535,37 @@ uint32_t *arm_syscall(uint32_t *regs)
         break;
     }
 
+#ifdef CONFIG_ARCH_ADDRENV
+  /* Check for a context switch.  If a context switch occurred, then
+   * CURRENT_REGS will have a different value than it did on entry.  If an
+   * interrupt level context switch has occurred, then establish the correct
+   * address environment before returning from the interrupt.
+   */
+
+  if (regs != CURRENT_REGS)
```

Comment:
armv7-r doesn't have MMU, so we don't need switch addrenv here.

Reply:
Done